app/components/page_list_component/view.rb

```diff
@@ -15,5 +15,9 @@ def show_up_button(index)
     def show_down_button(index)
       index != @pages.length - 1
     end
+
+    def question_text_for_page(id)
+      @pages.find { |page| page.id == id }.question_text
+    end
```

Comment:
Could we maybe add some unit tests for this method and the others above? These should be easier and faster to test than having to try find the results amongst the html thats rendered

Reply:
Done!